hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/slowlog/SlowLogEventHandler.java

```diff
@@ -0,0 +1,150 @@
+    this.slowLogPayloads = new SlowLogPayload[eventCount];
+  }
+
+  /**
+   * Called when a publisher has published an event to the {@link RingBuffer}
+   *
+   * @param event published to the {@link RingBuffer}
+   * @param sequence of the event being processed
+   * @param endOfBatch flag to indicate if this is the last event in a batch from
+   *   the {@link RingBuffer}
+   * @throws Exception if the EventHandler would like the exception handled further up the chain
+   */
+  @Override
+  public void onEvent(RingBufferEnvelope event, long sequence, boolean endOfBatch)
+      throws Exception {
+    SlowLogPayload slowLogPayload = event.getPayload();
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("Received Slow Log Event. RingBuffer sequence: {}, isEndOfBatch: {}, " +
+          "Event Call: {}", sequence, endOfBatch,
+        slowLogPayload.getCallDetails());
+    }
+    try {
+      if (LOCK.tryLock(WRITE_LOCK_WAIT_SEC, TimeUnit.SECONDS)) {
+        try {
+          final int index =
+            slowLogPayloadIdx.getAndUpdate(val -> (val + 1) % slowLogPayloads.length);
+          slowLogPayloads[index] = slowLogPayload;
+        } finally {
+          LOCK.unlock();
+        }
+      } else {
+        LOG.warn("Failed to acquire write lock while processing SlowLogEvent from RingBuffer");
+      }
+    } catch (InterruptedException e) {
+      LOG.warn("Failed to acquire write lock while processing SlowLogEvent from RingBuffer", e);
+    }
+  }
+
+  /**
+   * Cleans up slow log payloads
+   *
+   * @return true if slow log payloads are cleaned up, false otherwise
+   */
+  boolean clearSlowLogs() {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Received request to clean up online slowlog buffer..");
+    }
+    boolean isCleanedUp = false;
+    try {
+      if (LOCK.tryLock(WRITE_LOCK_WAIT_SEC, TimeUnit.SECONDS)) {
+        try {
+          Arrays.fill(slowLogPayloads, null);
+          slowLogPayloadIdx.set(0);
+          isCleanedUp = true;
+        } finally {
+          LOCK.unlock();
+        }
+      } else {
+        LOG.warn("Failed to acquire write lock while clearing slow logs");
+      }
+    } catch (InterruptedException e) {
+      LOG.warn("Failed to acquire write lock while clearing slow logs", e);
+    }
+    return isCleanedUp;
+  }
+
+  /**
+   * Retrieve list of slow log payloads
+   *
+   * @return list of slow log payloads
+   */
+  List<SlowLogPayload> getSlowLogPayloads() {
+    List<SlowLogPayload> slowLogPayloadList = null;
+    try {
+      if (LOCK.tryLock(READ_LOCK_WAIT_SEC, TimeUnit.SECONDS)) {
+        try {
+          slowLogPayloadList = Arrays.stream(slowLogPayloads)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        } finally {
+          LOCK.unlock();
+        }
+      } else {
+        LOG.warn("Failed to acquire read lock while retrieving slow logs");
+      }
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
```

Comment:
I'm not sure that throwing as a RTE is the right choice either. I think what you want to do is, at the very least, re-set the interrupt status on the current thread. Or you could modify your interface to include `throws InterruptedException` and leave it up to the caller to handle properly.